extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -0,0 +1,267 @@
+        } catch (err) {
+            this._logger.error('An error occured while getting old connectors', {
+                method: 'ConnectorsManager._getOldConnectors',
+                error: err.description || err.message,
+            });
+            throw errors.InternalError.customizeDescription(err.description);
+        }
+    }
+
+    /**
+     * Initialize previously created connector instances and
+     * creates new connectors based on configuration
+     * @returns {Promise|Connector[]} list connectors
+     * @throws {InternalError}
+     */
+    async initializeConnectors() {
+        try {
+            // get and initialize old connectors
+            const oldConnectorNames = await this._kafkaConnect.getConnectors();
+            if (oldConnectorNames) {
+                const oldConnectors = await this._getOldConnectors(oldConnectorNames);
+                this._connectors.push(...oldConnectors);
+                this._oldConnectors.push(...oldConnectors);
+            }
+            // Add connectors if required number of connectors not reached
+            const nbConnectorsToAdd = this._nbConnectors - this._connectors.length;
+            if (nbConnectorsToAdd > 0) {
+                for (let i = 0; i < nbConnectorsToAdd; i++) {
+                    // eslint-disable-next-line no-await-in-loop
+                    const newConnector = await this.addConnector();
+                    this._connectors.push(newConnector);
+                }
+            }
+            return this._connectors;
+        } catch (err) {
+            this._logger.error('An error occured while initializing connectors', {
+                method: 'ConnectorsManager.initializeConnectors',
+                error: err.description || err.message,
+            });
+            throw errors.InternalError.customizeDescription(err.description);
+        }
+    }
+
+    /**
+     * Removes invalid buckets from connector config
+     * @param {Connector} connector connector
+     * @param {string[]} buckets valid bucket names
+     * @returns {Promise|undefined} undefined
+     * @throws {InternalError}
+     */
+    async removeConnectorInvalidBuckets(connector, buckets) {
+        try {
+            // getting connector's invalid buckets
+            const invalidBuckets = connector.buckets.filter(bucket =>
+                !buckets.includes(bucket));
+            if (invalidBuckets.length > 0) {
+                // removing invalid buckets
+                await Promise.all(invalidBuckets.map(bucket =>
+                    connector.removeBucket(bucket, false)));
+                // updating connector pipeline
+                await connector.updatePipeline();
+            }
+        } catch (err) {
+            this._logger.error('An error occured while removing invalid buckets from a connector', {
+                method: 'ConnectorsManager.removeConnectorInvalidBuckets',
+                error: err.description || err.message,
+                connector,
```

Comment:
How is the connector going to be logged? I think it's better to just log its name for example, to avoid potentially a giant log line with a lot of internal fields logged.